docs/design-documents/docker_management/README.md

```diff
@@ -0,0 +1,151 @@
+
+**Prune docker image**
+docker image in temporary area is pruned by this workflow. Number of dates since updated is used as criteria for pruning.
+
+### Pipeline
+
+docker image management follows typically CI pipeline of build, test, release follows.
+
+There are some details worth mentioning though.
+
+**Build**
+
+docker buildx command is used for creating multi architecture docker image. To build docker image using buildx, one needs to push the image to a remote repository while building it. Since, we need to "test" before release, the resulting images are pushed to a temporary docker repository just after building.
+
+github container registry doesn't implement yet all the docker manifest APIs. Hence, a few features like deleting tag from an image is not available yet. Once these are implemented, we may remove temporary repository and add temporary tags to the image and delete the temporary tags after workflow.
+
+**Test**
+Once temporary images are built, we need to verify whether image built is same as the image already available as our intended docker image version. This is achieved by comparing the docker digest of two images. Obviously, if digest is same, there is no need to make another release as docker image available as released version is already up to date.
+
+If the digest is different, (means, new version of docker image is built), test.sh script takes care of all the testing.
+
+**Deploy**
+In Deploy job, depending on the result of test job, temporary images are moved to production repository.
+
+### Docker repository
+github provides free docker image storage for public repositories in github packages. The workflows make use of {{ secrets.GITHUB_TOKEN }}  https://docs.github.com/en/actions/reference/authentication-in-a-workflow 
+
+The packages are directly visible in mbed-os repository.
+
+For deleting images from temporary repository, a new token GITHUB_DELETE_IMAGE_TOKEN needs to be added with package delete permissions.
+
+There are two repositories being created or used for managing docker image 
+
+**mbed-os-env** This is the public repositories user can pull the images from. 
+
+**mbed-os-env-tmp** This is a private repositories created for temporary management of docker images while docker images are being tested. This is a staging area. Unfortunately, github container registry do no support renaming docker tags, hence the need for a temporary repository.
+
+### Workflow for forks
+
+As there is a scheduled trigger of workflow for development and release images, which inturn creates and updated docker images in github container registries, workflows are enabled only when "ARMMbed" repository owner criteria is met. For development, one can change this, make necessary development and put this back to "ARMMbed" when creating pull request to "ARMMbed/mbed-os".
+
+### Usage in example application
+![example app](./diagrams/example-app-usage.png)
+Example application repository mainly has three branches
```

Comment:
this section is covered in our workflow, do we need to maintain this one or just point to our workflow (not certain we have branching explained but we should add it there if not)

Reply:
I added this for completeness to the documentation. Idea was just to give a quick reference. I am happy to link it, if you can give me the link.

Reply:
https://github.com/ARMmbed/mbed-os-5-docs/issues/1472 - I created this one.  We can just use https://os.mbed.com/docs/mbed-os/latest/introduction/versions-and-releases.html here and this section can be then replaced by just reference and lets update our docs - I would like to have these information there.

Reply:
ok, will update and point to the link